### DISCUSSION TITLE
User-defined generator-notice?

### DISCUSSION THREAD

**Discussion by hellt** [squidfunk/mkdocs-material#4335] [Ideas]
I'd love to see insiders version to allow users to specify their own generator notice, instead of being able to just hiding it.

For example, it would be awesome to expose an option like `footer-note` or smth like that that would be able to take in a string with an icon expansion:

```
footer-note: Made with :octicons-heart-fill-24:{ .mdx-heart } by the [docs team](someurl).
``` 

<img width="335" alt="image" src="https://user-images.githubusercontent.com/5679861/189347539-97674f4e-2c94-4702-8108-d5fed4eaa121.png">

I think it adds another customization that makes insiders having its own personal touch. Overriding a partial is a viable option, but it gives me the anxiety to check each time if a footer partial didn't change with a new material release.

**Comment by squidfunk** [comment]
I like the idea. We could allow the `extra.generator` setting to be passed an HTML string, which is then used directly instead of the default generator notice. This has the limitation of not being able to use icons, only simple markup, though. For icons, we would need to pass everything through the Markdown parser, which is not possible without adding another plugin. That'll be overkill. However, for the simple use case, we can add the overriding functionality.

If that works for you, you can open a feature request.

**Reply by squidfunk** [reply]
Note that adding icons or custom markup is still possible by overriding the `copyright` partial. That's, in fact, already possible.

**Reply by hellt** [reply]
yes, I just realised that the copyright partial contains just the HTML for the generator notice. This worked for me.
For most users, though, I think the setting would be a nice addition to the insiders menu
